spec/javascripts/app/views/location_stream_spec.js

```diff
@@ -0,0 +1,39 @@
+describe("app.views.LocationMap", function() {
+  beforeEach(function(){
+    this.post = factory.post();
+    this.view = new app.views.LocationStream({model : this.post});
+  });
+
+  describe("toggleMap", function() {
+    context("with location provided", function() {
+      beforeEach(function(){
+        this.post.set({location : factory.location()}); // set location
+        gon.appConfig = { map: {enabled: true, mapbox: {enabled: true, id: "yourID", accessToken: "yourAccessToken" }} };
+        this.view.render();
+      });
+
+      it("should contain a map container", function() {
+        expect(this.view.$el[0]).toContainElement(".mapContainer");
+      });
+
+      it("should initialize map", function() {
+        expect(this.view.$el.find(".mapContainer")[0]).toHaveClass("empty");
+        this.view.toggleMap();
+        expect(this.view.$el.find(".mapContainer")[0]).not.toHaveClass("empty");
+      });
+
+      it("should change display status on every click", function() {
```

Comment:
Wow, @svbergerem thanks for figuring that out :) 
Btw, how did you find that solution? Can we somewhere read anything about that spec.content construction?